File: Problems/Middle-of-the-Linked-List/Middle-of-the-Linked-List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head){
    struct ListNode* slow = head;
    int i = 0;
    while (head != NULL) {
        head = head->next;
        if (i++ % 2)
            slow = slow->next;
    }
    return slow;
}

File: Problems/Middle-of-the-Linked-List/Middle-of-the-Linked-List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let slow = head, i = 0;
    while (head) {
        head = head.next;
        if (i++ % 2)
            slow = slow.next;
    }
    return slow;
};
